actors/miner/tests/miner_actor_test_precommit_batch.rs

```diff
@@ -0,0 +1,317 @@
+    let total_deposit: BigInt = deposits.iter().sum();
+    let total_balance = net_fee + total_deposit.clone();
+    rt.set_balance(total_balance + balance_surplus);
+
+    if exit_code != ExitCode::OK {
+        expect_abort_contains_message(
+            exit_code,
+            error_str,
+            (|| {
+                h.pre_commit_sector_batch(
+                    &mut rt,
+                    PreCommitSectorBatchParams { sectors },
+                    &conf,
+                    base_fee,
+                )?;
+
+                // State untouched.
+                let st: State = rt.get_state();
+                assert!(st.pre_commit_deposits.is_zero());
+                let expirations = h.collect_precommit_expirations(&rt, &st);
+                assert!(expirations.is_empty());
+                Ok(())
+            })(),
+        );
+        return;
+    }
+    let precommits = h.pre_commit_sector_batch_and_get(
+        &mut rt,
+        PreCommitSectorBatchParams { sectors: sectors.clone() },
+        &conf,
+        base_fee,
+    );
+
+    // Check precommits
+    let st: State = rt.get_state();
+    for i in (0..batch_size).map(|i| i as usize) {
+        assert_eq!(precommit_epoch, precommits[i].pre_commit_epoch);
+        assert_eq!(conf.sector_weights[i].deal_weight, precommits[i].deal_weight);
+        assert_eq!(conf.sector_weights[i].verified_deal_weight, precommits[i].verified_deal_weight);
+
+        assert_eq!(sector_nos[i], precommits[i].info.sector_number);
+        assert_eq!(sectors[i], precommits[i].info);
+
+        let pwr_estimate = qa_power_for_weight(
+            h.sector_size,
+            sectors[i].expiration - precommit_epoch,
+            &conf.sector_weights[i].deal_weight,
+            &conf.sector_weights[i].verified_deal_weight,
+        );
+        let expect_deposit = pre_commit_deposit_for_power(
+            &h.epoch_reward_smooth,
+            &h.epoch_qa_power_smooth,
+            &pwr_estimate,
+        );
+        assert_eq!(expect_deposit, precommits[i].pre_commit_deposit);
+    }
+
+    assert!(total_deposit.is_positive());
+    assert_eq!(total_deposit, st.pre_commit_deposits);
+
+    let expirations = h.collect_precommit_expirations(&rt, &st);
+    let expected_precommit_expiration = st.quant_spec_every_deadline(&rt.policy).quantize_up(
+        precommit_epoch
+            + rt.policy.max_prove_commit_duration[&h.seal_proof_type]
+            + rt.policy.expired_pre_commit_clean_up_delay,
+    );
+    assert_eq!(HashMap::from([(expected_precommit_expiration, sector_no_as_uints)]), expirations);
+}
+
+mod miner_actor_precommit_batch {
+    use super::*;
+
+    #[test]
+    fn one_sector() {
+        assert_simple_batch(1, TokenAmount::zero(), TokenAmount::zero(), &[], ExitCode::OK, "");
+    }
+
+    #[test]
+    fn max_sectors() {
+        assert_simple_batch(32, TokenAmount::zero(), TokenAmount::zero(), &[], ExitCode::OK, "");
+    }
```

Comment:
The max changed to 256 between this test and the actual upgrade.  Please call this one thirty_two_sectors and then add another one using the policy/miner constant for max.